File: RecuperacionGestionEmpresarial/Domain/Interfaces/IDepartamentoUseCase.cs
using Domain.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain.Interfaces
{
    /// <summary>
    /// Interfaz del caso de uso para gestionar la lógica de negocio de departamentos
    /// </summary>
    public interface IDepartamentoUseCase
    {
        /// <summary>
        /// Devuelve una lista con todos los departamentos
        /// </summary>
        /// <returns>Lista de departamentos</returns>
        public List<Departamento> getDepartamentos();
    }
}

File: Tema15/ListadoPersonasAngular/src/Domain/Interfaces/IDepartamentoUseCase.ts
import { Departamento } from '../Entities/Departamento';

export interface IDepartamentoUseCase {
  getDepartamentos(): Promise<Departamento[]>;
  getDepartamento(id: number): Promise<Departamento>;
  agregarDepartamento(departamento: Departamento): Promise<number>;
  actualizarDepartamento(departamento: Departamento): Promise<number>;
  eliminarDepartamento(id: number): Promise<number>;
}